### PR TITLE
Add .coafile and fix styling

### PR DESCRIPTION
**Pull request by andrewda**
This adds code linting using a .coafile which checks the
ESLintBear, SpaceConsistencyBear, LineLengthBear and
InvalidLinkBear.

Closes https://github.com/coala/gci-leaders/issues/11

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "parserOptions": {
+    "ecmaVersion": 2017
+  },
+  "extends": [
+    "prettier"
+  ],
+  "rules": {
+    "semi": [ "error", "never" ],
+    "indent": [ "error", 2 ],
+    "comma-dangle": [ "error", "always-multiline" ]
```
Comment (blazeu) [inline .eslintrc]: Why not `prettier/prettier` ?

File: package.json [hunk 2]
```diff
@@ -12,5 +12,9 @@
   "dependencies": {
     "mustache": "^2.3.0",
     "node-fetch": "^1.7.3"
+  },
+  "devDependencies": {
+    "eslint": "^4.12.1",
+    "eslint-config-prettier": "^2.9.0"
```
Comment (blazeu) [inline package.json]: eslint-config-prettier is for turning off eslint rules that conflict with prettier

https://github.com/prettier/eslint-plugin-prettier should be used instead

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+before_install:
+  - nvm install 8
+  - python3 --version
+  - node --version
+  - npm install
+  - npm install -g eslint prettier eslint-plugin-prettier
+  - pip3 install coala-bears
 
 script:
+  - prettier --single-quote --no-semi --trailing-comma=es5 --write
```
Comment (blazeu) [inline .travis.yml]: the CI doesn't need autofixing

File: .travis.yml [hunk 4]
```diff
@@ -1,7 +1,15 @@
-sudo: false
-language: node_js
+sudo: required
+language: python
+python: 3.6
+
+before_install:
+  - nvm install 8
+  - npm install
+  - npm install -g eslint prettier eslint-plugin-prettier
```
Comment (blazeu) [inline .travis.yml]: Why?

Comment (andrewda) [reply .travis.yml]: https://travis-ci.org/coala/gci-leaders/builds/312179396#L748

File: .travis.yml [hunk 5]
```diff
@@ -1,7 +1,15 @@
-sudo: false
-language: node_js
+sudo: required
```
Comment (jayvdb) [inline .travis.yml]: why did you need sudo ?

File: .eslintrc [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
+  "parserOptions": {
+    "ecmaVersion": 2017
+  },
+  "plugins": [
+    "prettier"
+  ],
+  "rules": {
+    "prettier/prettier": ["error", {
```
Comment (jayvdb) [inline .eslintrc]: JSON checking of this file please
and if it supports yaml config, that would be preferred

Comment (blazeu) [reply .eslintrc]: Yes eslint does support yaml config

File: .travis.yml (hunk 4 above)
Comment (blazeu) [reply .travis.yml]: you can add `node_modules/.bin` to PATH
and set NODE_PATH (see our dockerfile)

https://github.com/coala/docker-coala-base/blob/master/Dockerfile#L10
https://github.com/coala/coala-atom/blob/master/circle.yml#L32

**Comment by blazeu** [comment]
I know I said to ignore json file, but can we lint package.json using JSONBear
and make sure prettier/eslint doesn't run there, it can conflict with npm's formatting.

**Review by blazeu — CHANGES_REQUESTED** [review]
Still missing YAMLLintBear.

File: package.json [hunk 7]
```diff
@@ -1,16 +1,21 @@
...
-  "scripts": {
-    "build": "node scrap.js && node generate.js"
-  },
-  "keywords": [],
-  "author": "blazeu <neversleepman@tfwno.gf>",
-  "license": "MIT",
-  "dependencies": {
-    "mustache": "^2.3.0",
-    "node-fetch": "^1.7.3"
-  }
```
Comment (blazeu) [inline package.json]: Keep the indentation from npm (2 spaces).

I know npm@5 does read the indentation and use that,
but if we keep the previous style, we don't need to modify this file :)

**Review by blazeu — DISMISSED** [review]

File: .travis.yml [hunk 8]
```diff
@@ -1,13 +1,21 @@
...
   - npm run build
 
 deploy:
   provider: pages
   skip_cleanup: true
-  github_token: $GITHUB_TOKEN # Set in travis-ci.org dashboard
+  github_token: $GITHUB_TOKEN
   local_dir: out
-  on:
+  "on":
```
Comment (jayvdb) [inline .travis.yml:20]: really?

Can you link to the build showing the bear/linter complaining about this?

Comment (andrewda) [reply .travis.yml:20]: Yea this was problematic...

```yml
.travis.yml
|  19| ··on:
|    | [NORMAL] YAMLLintBear:
|    | truthy value is not quoted (truthy)
```

I think can also make a .yamllintrc file to make this work, though, but it adds yet another lint config file to the repo.

**Comment by jayvdb** [comment]
ack https://github.com/coala/gci-leaders/pull/20/commits/dff8ba5094d165a821f60d6bcfcf3385c28d5bba

**Review by jayvdb — APPROVED** [review]
